packages/website/pages/docs/how-to/get-status.md

```diff
@@ -0,0 +1,82 @@
+# Getting upload status
+
+import Callout from 'nextra-theme-docs/callout';
+
+The NFT.Storage service provides a few ways to check on the status of an upload.
+
+On the [File listing page](https://nft.storage/files) of the NFT.Storage site, you can see a list of things that you've uploaded to NFT.Storage.
+
+![Screenshot of NFT.Storage file listing page, showing a table with an entry for each upload.](/images/nft-storage-file-listing.png)
+
+The table on the file listing page has two columns related to the current status of the data storage process, [Pin Status](#pin-status), and [Storage Providers](#storage-providers).
+
+The information in the file listing page is also available using the [JavaScript client's `check` method][client-js-check], and via the [HTTP API][client-http] by sending a `GET` request to the `/check/{cid}` endpoint.
+
+## Pin status
+
+When you upload data to NFT.Storage, it is ["pinned"][ipfs-docs-pinning] to several [IPFS clusters][ipfs-cluster], which are groups of machines that coordinate to provide content-addressed data to the [IPFS](https://ipfs.io) peer-to-peer network.
+
+The **Pin Status** column shows the status of the pinning operation. There are three possible values for the Pin Status column: `Queuing`, `Pinned`, and `Failed`.
+
+The `Queuing` status indicates that a request to pin the data has been submitted to a cluster node, and the node has added the request to a queue for pinning. Depending on how busy the cluster is, an upload may stay in the `Queuing` state for some time before transitioning to `Pinned` or `Failed`.
+
+A `Pinned` status means that the cluster has successfully pinned the data and will retain a copy until instructed otherwise.
+
+A `Failed` status indicates that an operation failed while attempting to pin the data. Note that a `Failed` status may be temporary, as the cluster will continually retry `Failed` pins in the background and will update the status to `Pinned` if a retry attempt is successful.
```

Comment:
Would leave a note that a Failed pin indicates that the initial upload attempt failed, and it will resolve back to Pinned if the data with the same CID was later successfully uploaded (though it might take a bit of time to update). If you think your upload is in this state, you can test this by retrieving your data from an HTTP gateway (if it's a directory, sample a set of files in the directory).

Reply:
you do cover some of this below, but would still float some of it up to the summary paragraph on failed pins here